programs/ssl/ssl_server2.c

```diff
@@ -2020,7 +2020,6 @@ int main( int argc, char *argv[] )
     if( received_sigterm )
     {
         mbedtls_printf( " interrupted by SIGTERM\n" );
-        ret = 0;
```

Comment:
There is another place in the code where `received_sigterm` is checked. Why ignore `ret` here, but reset it in the other place, after the call to `mbedtls_net_accept`?

I think that resetting `ret` can hide bugs in the server.

When `ret = 0` is removed from the post-signal-handler code after `mbedtls_net_accept`, the error under normal conditions is
```
  . Waiting for a remote connection ... interrupted by signal
Last error was: -0x4A - NET - Could not accept the incoming connection
```
This is caught by the “Default” tests which check that the server does not report any error.

I think that in this place, `ret` should be reset only if it was `MBEDTLS_ERR_NET_ACCEPT_FAILED`.

Reply:
Because `net_accept()` is the only place where we expect to be interrupted by a `SIGTERM` when running `ssl-opt.sh` and I chose to care only about that.

I didn't add the test to reset only if `ret` was `MBEDTLS_ERR_NET_ACCEPT_FAILED` because I thought it was the only possible return code of `net_accept()`, but I failed to check and now I realise I was wrong, at least with UDP it can return other things, so I'll add that check.